script.js

```diff
@@ -92,88 +87,18 @@ function searchWatches(element, inputValue) {
-      filteredArray.push(data[i]);
-  }
-  return displayWatches(filteredArray);
-}
-function getFullpaveBracelet(bracelet) {
-  clearContainer();
-  let filteredArray = [];
-  let fullpaveBraceletType = fullpaveBracelet.textContent.toLowerCase();
-  console.log(fullpaveBraceletType);
-
-  for (let i = 0; i < data.length; i++) {
-    if (data[i].bracelet.includes(fullpaveBraceletType))
-      filteredArray.push(data[i]);
-  }
-  return displayWatches(filteredArray);
-}
-function getMilaneseBracelet(bracelet) {
-  clearContainer();
-  let filteredArray = [];
-  let milaneseBraceletType = milaneseBracelet.textContent.toLowerCase();
-
-  for (let i = 0; i < data.length; i++) {
-    if (data[i].bracelet.includes(milaneseBraceletType))
-      filteredArray.push(data[i]);
-  }
-  return displayWatches(filteredArray);
-}
-function getMetalBracelet(bracelet) {
-  clearContainer();
-  let filteredArray = [];
-  let metalBraceletType = metalBracelet.textContent.toLowerCase();
-
-  for (let i = 0; i < data.length; i++) {
-    if (data[i].bracelet.includes(metalBraceletType))
-      filteredArray.push(data[i]);
-  }
-  return displayWatches(filteredArray);
-}
-
-// - - - tried using switch for bracelet types, but cannot finish it -----//
-const filterByBracelet = (event) => {
-  clearContainer();
-  const options = braceletOptions.childNodes; // returns children of braceletOptions: LIs, inside which are a tags, inside which there is a text with bracelet name. I dont know how to select the text - we need to put it inside braceletType
-  console.log(options);
-  let braceletType = options.textContent; // ???
-  console.log(braceletType); // currently shows undefined
-  let filteredArray = [];
-  switch (event.target.value) {
-    case "leather bracelet":
-      for (let i = 0; i < data.length; i++) {
-        if (data[i].bracelet.includes(braceletType))
-          filteredArray.push(data[i]);
-      }
-      break;
-    case "fullpave bracelet":
-      for (let i = 0; i < data.length; i++) {
-        if (data[i].bracelet.includes(braceletType))
-          filteredArray.push(data[i]);
-      }
-      break;
-    case "milanese bracelet":
-      for (let i = 0; i < data.length; i++) {
-        if (data[i].bracelet.includes(braceletType))
-          filteredArray.push(data[i]);
-      }
-      break;
-    case "metal bracelet":
-      for (let i = 0; i < data.length; i++) {
-        if (data[i].bracelet.includes(braceletType))
-          filteredArray.push(data[i]);
-      }
-      break;
-  }
-  return displayWatches(filteredArray);
-};
+ // Search by category(bracelet)
+ function searchByBracelet() {
+   const filteredArray = [];
+   const braceletCategory = this.id.split("-")[0]; // get clicked id and split it to get exact bracelet type
+   for(let i = 0; i < data.length; i++) {
+     if (data[i].bracelet.includes(braceletCategory)) {
+       filteredArray.push(data[i])
+     }
+   }
+   clearContainer();
+   return displayWatches(filteredArray);
```

Comment:
Actually, instead of lines 92~100 we could use higher-level function, like this:
`
return data.filter(watch => watch.includes(braceletCategory))
`